examples/WORKSPACE

```diff
@@ -32,6 +32,10 @@ py_repositories()
 
 python_register_toolchains(
     name = "rules_ros2_python",
+    # Note that it is required to pin this specific version because of dependencies to numpy headers from rules_ros2's 3.10 numpy dep.
+    # Otherwise you will get errors at runtime (not build time!) like:
+    #     ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
+    #     Importing the numpy C-extensions failed.
```

Comment:
Not quite accurate. We need first because the provided `requirements_lock.txt ` is resolved for Python 3.10. Then we need that because of numpy. It should be formulated that one needs to keep the python_version and the resolved lock file in sync using the same Python version.

Reply:
I guess you could say that yes. The strict dependency is caused by https://github.com/mvukov/rules_ros2/blob/92338a842b0ad959211419438d558474f9d4f2fa/ros2/BUILD.bazel#L94-L106 and https://github.com/mvukov/rules_ros2/blob/92338a842b0ad959211419438d558474f9d4f2fa/ros2/interfaces.bzl#L829-L835 which in turn depends on the requirements_lock.txt and current_py_cc_headers version to be in sync.

Reply:
rephrased.